### ISSUE TITLE
"composer update" fails for OroCommerce 1.6 application

### ISSUE THREAD

**Issue by aligentjim** [oroinc/orocommerce#64]
Using an unmodified version of the orocommerce-application repository I'm unable to run "composer update" due to dependency errors (see below).  We've also seen the same issue on an existing 1.5 install that we're upgrading to 1.6.  In that case our normal procedure would be to merge changes in `composer.json` and run `composer update` to fix the `composer.lock` file.  

It looks like the "out of the box" composer.json requires a set of packages which is no longer installable.

The composer output is shown below...

```
$ composer update --no-dev -v
VESSEL: vessel user will have id: 775000001
Loading composer repositories with package information
Updating dependencies
Dependency resolution completed in 109.278 seconds
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Conclusion: don't install phpunit/phpunit 5.7.27
    - Conclusion: remove phpunit/phpunit 5.7.26
    - Conclusion: don't install phpunit/phpunit 5.7.26
    - Conclusion: don't install phpunit/phpunit 5.7.25
    - Conclusion: don't install phpunit/phpunit 5.7.24
    - Conclusion: don't install phpunit/phpunit 5.7.23
    - Conclusion: don't install phpunit/phpunit 5.7.22
    - Conclusion: don't install phpunit/phpunit 5.7.21
    - Conclusion: don't install phpunit/phpunit 5.7.20
    - Conclusion: don't install phpunit/phpunit 5.7.19
    - Conclusion: don't install phpunit/phpunit 5.7.18
    - Conclusion: don't install phpunit/phpunit 5.7.17
    - Conclusion: don't install phpunit/phpunit 5.7.16
    - Conclusion: don't install phpunit/phpunit 5.7.15
    - Conclusion: don't install phpunit/phpunit 5.7.14
    - Conclusion: don't install phpunit/phpunit 5.7.13
    - Conclusion: don't install phpunit/phpunit 5.7.12
    - Conclusion: don't install phpunit/phpunit 5.7.11
    - Conclusion: don't install phpunit/phpunit 5.7.10
    - Conclusion: don't install phpunit/phpunit 5.7.9
    - Conclusion: don't install phpunit/phpunit 5.7.8
    - Conclusion: don't install phpunit/phpunit 5.7.7
    - Conclusion: don't install phpunit/phpunit 5.7.6
    - Conclusion: don't install phpunit/phpunit 5.7.5
    - Conclusion: don't install phpunit/phpunit 5.7.4
    - Conclusion: don't install phpunit/phpunit 5.7.3
    - Conclusion: don't install phpunit/phpunit 5.7.2
    - Conclusion: don't install phpunit/phpunit 5.7.1
    - Conclusion: don't install phpunit/phpunit 5.7.0
    - Conclusion: remove phpunit/php-timer 1.0.9
    - Installation request for phpunit/phpunit 5.7.* -> satisfiable by phpunit/phpunit[5.7.0, 5.7.1, 5.7.10, 5.7.11, 5.7.12, 5.7.13, 5.7.14, 5.7.15, 5.7.16, 5.7.17, 5.7.18, 5.7.19, 5.7.2, 5.7.20, 5.7.21, 5.7.22, 5.7.23, 5.7.24, 5.7.25, 5.7.26, 5.7.27, 5.7.3, 5.7.4, 5.7.5, 5.7.6, 5.7.7, 5.7.8, 5.7.9, 5.7.x-dev].
    - Conclusion: remove sebastian/phpcpd dev-master
    - sebastian/phpcpd dev-master requires phpunit/php-timer ^2.0 -> satisfiable by phpunit/php-timer[2.0.0, 2.0.x-dev].
    - Can only install one of: phpunit/php-timer[2.0.0, 1.0.x-dev].
    - Can only install one of: phpunit/php-timer[2.0.x-dev, 1.0.x-dev].
    - phpunit/phpunit 5.7.x-dev requires phpunit/php-timer ^1.0.6 -> satisfiable by phpunit/php-timer[1.0.9, 1.0.6, 1.0.7, 1.0.8, 1.0.x-dev].
    - Conclusion: don't install phpunit/php-timer 1.0.9|don't install sebastian/phpcpd dev-master
    - Conclusion: don't install phpunit/php-timer 1.0.8|don't install sebastian/phpcpd dev-master
    - Conclusion: don't install phpunit/php-timer 1.0.7|don't install sebastian/phpcpd dev-master
    - Conclusion: don't install phpunit/php-timer 1.0.6|don't install sebastian/phpcpd dev-master
    - Installation request for sebastian/phpcpd dev-master#cff7f36c2ae89d59987de25d14fd41a72dd4a703 as 3.1.0 -> satisfiable by sebastian/phpcpd[dev-master].

Running update with --no-dev does not mean require-dev is ignored, it just means the packages will not be installed. If dev requirements are blocking the update you have to resolve those problems.

```

**Comment by aivus** [comment]
Hi @aligentjim.

Thank you for your issue.

As I see you already contacted with us via support desk.
We will give you answer there in ticket OPS-945

**Comment by alexbaat** [comment]
Is there any solution for this?

**Comment by aivus** [comment]
Hi @alexbaat 

Fix should be released in the next patch version (1.6.1)
